flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -133,8 +134,13 @@
 	 * Input channels. There is a one input channel for each consumed intermediate result partition.
 	 * We store this in a map for runtime updates of single channels.
 	 */
+	@GuardedBy("requestLock")
 	private final Map<IntermediateResultPartitionID, InputChannel> inputChannels;
 
+	/** A mapping from internal channel index in this gate to input channel. */
+	@GuardedBy("requestLock")
+	private final Map<Integer, InputChannel> indexToInputChannelMap;
```

Comment:
ok

Reply:
Actually, this could be a simple array, couldn't it? If you look at `org.apache.flink.runtime.io.network.partition.consumer.SingleInputGate#create` you'll see that `0 <= channelIndex < inputChannels.length`

Reply:
and maybe call it `inputChannelsByChannelIndex` instead?